rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+---
+openapi: 3.0.3
+info:
+  title: Apache Iceberg REST Catalog API
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+  description:
+    Defines the specification for the first version of the REST Catalog API. Implementations should support both Iceberg table specs v1 and v2, with priority given to v2.
+servers:
+  - url: https://{host}:{port}/{basePath}
+    variables:
+      host:
+        description: The host address for the specified server
+        default: localhost
+      port:
+        description: The port used when addressing the host
+        default: "443"
+      basePath:
+        default: v1
+  - url: http://127.0.0.1:1080/v1
+    description: URL Used for Mock-Server Unit Tests
+# All routes are currently configured using an Authorization header.
+security:
+  - BearerAuth: []
+paths:
+  /config:
+    get:
+      tags:
+        - Configuration API
+      summary: List all catalog configuration settings
+      operationId: getConfig
+      description: >
+        All REST catalog clients will first call this route to get possible catalog-specific
+        configuration values provided by the server, that the catalog (and its HTTP client)
+        can use to complete the `initialize` step.
+
+        This call is similar to the initial set-up calls that some catalogs already do for
+        domain-specific information, such as the Nessie catalog or the Glue catalog.
+        This is to allow for services that would like to integrate with Iceberg to do so,
+        and to be able to add their own domain-specific information into the REST catalog without
+        requiring them to write and distribute a catalog themselves.
+
+        There will be two sets of values provided -
+
+        - overrides
+          * An object containing values that the client must use.
+            For example, auth headers that the client will receive from the server
+            as temporary credentials.
+        - defaults
+          * Catalog-specific configuration that the client may use as a default value.
+            These are optional and the client is free to use its own value for these.
```

Comment:
I'd suggest using the terms `normative` and `optional` instead. It is not clear what the "overrides" are overriding since the intention is obviously for these values to be the source of truth.

Reply:
By the way, what is a use case for the "default" properties? Could you add an example, please?

Reply:
> I'd suggest using the terms normative and optional instead. It is not clear what the "overrides" are overriding since the intention is obviously for these values to be the source of truth.

For me, `normative` is rather confusing. The word `overrides` was used meaning that it would override any client supplied value as it's the source of truth (e.g. it's required).

I'm open to using a different word, but I haven't typically heard `normative` used in this sense and I think it would cause more confusion for end users than choosing some other word.

Reply:
Well, if we approach it from the other end (to try and understand this better), what is the intended use case for `defaults`?